### ISSUE TITLE
A list of raised hands/requests to speak

### ISSUE THREAD

**Issue by forteller** [jitsi/jitsi-meet#8104]
**Is your feature request related to a problem you are facing?**
In my experience lots of moderators in video meetings does not notice the raised hand feature at all. And if they do, they don't remember who asked to speak first or second or third, etc. To keep track of this they have to maintain a list of who has asked to speak.

**Describe the solution you'd like**
Since Jitsi already knows who raised their hand when, Jitsi should maintain that list automatically. The first person to raise their hand is added to the top, and the list grows from there. After the top person has spoken, their name is removed from the list. 

This will have many positive effects:
- It will be much harder for the moderator to not notice the raised hand
- The moderator will know exactly in what order to allow who to speak
- The other participants could also see who's requested to speak

Possible enhancements of the feature (nice to have in the future, not need to have for the feature to be useful):
- From the list it could also be possible to:
   - ask to reply to the current speaker (you name would be added directly under the current speaker (the top name of the list), a little bit indented)
   - remove yourself from the queue
   - perhaps send a meeting card to the current speaker, see #7239
- Set a timer for how long each speaker (and each reply) gets to speak, and show the timer beside their name. 
   - They could be muted automatically when the time is up. This should be optional, of course.
- Could add an option to mute everyone who isn't at the top of the list.

This feature request could also solve #8089

**Describe alternatives you've considered**
Alternative is that the moderator keeps a list manually, but then they have to notice your raised hand first, which they often don't. Another alternative is that each person wanting to speak asks in the chat, rather than with the raise hand feature, but if the chat is used for other things, then it could be hard to follow.

Thanks a ton for the amazing, and important work you do with this software!

**Comment by doobilydo** [comment]
Keeping a "stack". This is being used in calls/meetings. It would make a lot of sense to make it a feature of the software.

A "stack" button to automatically add/remove users to/from a visible list might be helpful. I think it makes sense to mute all others until their turn.

https://en.wikipedia.org/wiki/Progressive_stack (regular stack)
https://en.wikipedia.org/wiki/Queue_area

**Comment by forteller** [comment]
I've been told that Zoom does this by putting the names in the list of all participants in the order of who used the raised hand feature first. This could be the way to do it. But Jitsi doesn't even show the raised hand in the list of participants at all. Also, you can only show the list on top of everything else. 

So a good way to start working on this feature might be to let you see a list of participants in a sidebar, and then showing raised hand in that list?

**Comment by forteller** [comment]
I believe that this feature request deserves to live on, because I am truly convinced that it is an important feature that needs implementing for Jitsi to be a good alternative to the proprietary services.

Also, it seems @m0hdasif has started working on it?

**Comment by rngadam** [comment]
This just came up in a meeting with a lot of raised hand. became impossible to track who raised their hands.... please reopen or reference to superceding issue

**Comment by saghul** [comment]
This is already implemented. By default the participants pane is alphabetically sorted:
<img width="2032" alt="Screenshot 2021-09-23 at 10 44 19" src="https://user-images.githubusercontent.com/317464/134478502-abf53f1c-66b6-48a0-9c9d-49d557df6310.png">

But if participants raise their hand they'll be moved to the top:

<img width="2032" alt="Screenshot 2021-09-23 at 10 44 41" src="https://user-images.githubusercontent.com/317464/134478574-e1d24e80-e67d-48e8-9ff4-02bc679ee7a0.png">

This has recently been deployed to meet.jit.si, it will soon be available in a stable release.

**Comment by forteller** [comment]
That's great! Thank you! Are they sorted chronologically, by who raised their hand first, or alphabetically?

**Comment by saghul** [comment]
Chronologically, but the information is not propagated through the backend so participants joining late won't necessarily see the same order. That will come in a further patch.

**Comment by rngadam** [comment]
> This has recently been deployed to meet.jit.si, it will soon be available in a stable release.

Could I suggest attaching issues that are "closed" to a planned release version? Maybe by creating milestones and attaching these issues that are fixed but haven't been released yet? To me "closed" would mean that it's been fully tested, deployed and made available in the source.

**Comment by albjeremias** [comment]
I had a bad experience with this setup. Because I am the first in the room, and i saw the raise hand with 2.. it was really strange i didn't understand that was the hands of all the users in the room, maybe its misplaced? or is just an android xperience?

**Comment by saghul** [comment]
Not sure what you mean. Can you share a screenshot?

**Comment by cybergon** [comment]
I'm not sure if this team is the same one developing the phone app, but regarding this

> But if participants raise their hand they'll be moved to the top:

it doesn't work on the app.

**Comment by saghul** [comment]
It is us, same team. Since this issue started 3 years ago, can you please open a new one with the mobile specific problem? Thank you!
